typescript/src/schema/naming.ts

```diff
@@ -0,0 +1,68 @@
+import * as plugin from '../../../pbjs-genfiles/plugin';
+import {commonPrefix} from '../util';
+
+export class Naming {
+  name: string;
+  namespace: string[];
+  version: string;
+  productName: string;
+  protoPackage: string;
+
+  constructor(fileDescriptors: plugin.google.protobuf.IFileDescriptorProto[]) {
+    const protoPackages = fileDescriptors.map(fd => fd.package || '');
+    const rootPackage = commonPrefix(protoPackages).replace(/\.$/, '');
+    if (!rootPackage) {
+      throw new Error('Protos provided have different proto packages.');
+    }
+
+    const pattern =
```

Comment:
Could you add a comment so we know what this regex does?

Reply:
```suggestion
    // Define the regular expression to match a version component
    // (e.g. "v1", "v1beta4", etc.).
    const pattern =
```